addon/hint/show-hint.js

```diff
@@ -400,7 +400,8 @@
     var cur = cm.getCursor(), token = cm.getTokenAt(cur);
     var to = CodeMirror.Pos(cur.line, token.end);
     if (token.string && /\w/.test(token.string[token.string.length - 1])) {
```

Comment:
Maybe this test should also check for the character before the cursor now, rather than the one at the end of the token?

Reply:
Do you have an example? I'm not quite getting what you mean :D

Reply:
I meant the test right before your change, where it only uses the token text when it ends in a word character.

Reply:
I saw the line, but didn't understand why you'd want to check for the character in front of the cursor.
So effectively, that would skip completions for tokens containing non-word characters, if the cursor is right after one?

Actually, I'd even remove the current regex test and let the word check loop handle everything. As you said once [here](https://github.com/codemirror/CodeMirror/issues/5169#issuecomment-356540344), being able to put anything into the word list may be useful. For example if you work with AngularJS, var and function names starting with `$` could be completed right at the first character.

Reply:
That's a good point. But then we'd want to handle the case where no matches are found _and_ the cursor is directly after a non-word character specially, by also trying to complete with the empty string, so that you can still complete (usual, word completions) with the cursor after a non-word token.

Reply:
Ah I see, you mean something like completions after a dot for example. Since the dot would be considered as a valid token, the completion list would be empty.

In that case maybe this could solve the problem `if (token.string && (token.type == 'variable' || token.type == 'property'))`. A third OR condition would be needed if the current mode doesn't support these token types. Also this would make the addon complete in less situations, which may be good or bad, depending on your situation.
Now that I think about it, we maybe should make this configurable (array of valid token types), then the power lies in the users hand and the addon can be fine tuned for each use case which would make it way more practical.
But that's just an idea and out of scope for this pull request. If you still want the word check you suggested initially, I can implement that for now, even though I don't think it would add anything because we are inside one token which ends with a word character.